core/src/wsv.rs

```diff
@@ -596,21 +555,15 @@ impl WorldStateView {
 
     /// Construct [`WorldStateView`] with specific [`Configuration`].
     #[inline]
-    pub fn from_configuration(
-        config: Configuration,
-        world: World,
-        events_sender: EventsSender,
-    ) -> Self {
-        let (new_block_notifier, _) = tokio::sync::watch::channel(());
-
+    pub fn from_configuration(config: Configuration, world: World) -> Self {
         Self {
             world,
             config,
             transactions: DashSet::new(),
             blocks: Arc::new(Chain::new()),
-            metrics: Arc::new(Metrics::default()),
-            new_block_notifier: Arc::new(new_block_notifier),
-            events_sender,
+            events_buffer: std::cell::RefCell::new(Vec::new()),
+            metric_tx_amounts: std::cell::Cell::new(0.0_f64),
+            metric_tx_amounts_counter: std::cell::Cell::new(0),
```

Comment:
Doesn't look like we fully decoupled the behaviours. But I think it's OK for this PR.

Reply:
Just idea but maybe metrics could be updated through events? Such approach seems to decouple metrics.

Reply:
No. That would basically the same as we had before. It's not strange that you can't fully decouple metrics from the thing they measure. The point is making the contents of the WSV less global in nature and more just data.

Reply:
> That would basically the same as we had before. 
In which way?